runtime/src/main/java/org/corfudb/runtime/ViewsGarbageCollector.java

```diff
@@ -83,8 +84,7 @@ public void runRuntimeGC() {
                     endTs - startTs, runtime.getObjectsView().getObjectCache().size(), trimMark);
         } catch (Exception e) {
             if (e.getCause() instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-                throw new GarbageCollectorException(e);
```

Comment:
Throw GarbageCollectorException instead of UnrecoverableCorfuInterruptedError?

Reply:
I would like the client to have only one un-checked exception type that signals interrupts. Is there a specific reason why GarbageCollectorException was proffered here? Also looks like this is the sole usage of GarbageCollectorException.

Reply:
I see.

> Is there a specific reason why GarbageCollectorException was proffered here? 
I'm not sure.

Just make sure you know that UnrecoverableCorfuInterruptedError is an Error which would require you to catch a throwable now on the client side code, while GarbageCollectorException is an Exception.